util/io/game_state.py

```diff
@@ -22,10 +26,24 @@ def __init__(self, packet: GameTickPacket, previous_game_state, index: int):
         self.team = self.car.team
         self.ball = BallData(packet.game_ball.physics)
 
+    def toFrameworkGameState(self):
+        ball_state = framework_utils.BallState(
+            framework_utils.Physics(location=self.ball.position.to_game_state_vector3(),
+                                    velocity=self.ball.velocity.to_game_state_vector3()))
+        return framework_utils.GameState(ball=ball_state)
+
     @staticmethod
     def __generateHumanCarList(car_list: List[CarData]):
         human_car_list = []
         for car in car_list:
             if not car.is_bot:
                 human_car_list.append(car)
         return human_car_list
+
+    @classmethod
+    def toJSON(cls, game_state):
+        return jsonpickle.encode(game_state)
+
+    @classmethod
+    def fromJSON(cls, received_sync_data):
+        return jsonpickle.decode(received_sync_data)

```

Comment:
moved the json formating INTO the class, instead of doing this in some random remote class